File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] stock) {
        
       int maxProfit=0;
       int minStock = stock[0];
       
       for(int i=1;i<stock.length;i++){
           int profit=0;
           if(minStock>stock[i]){
               minStock=stock[i];
           }
           if(stock[i]>minStock){
               profit=stock[i]-minStock;
           }
           if(profit>maxProfit){
               maxProfit=profit;
           }
        
            
       }
       return maxProfit;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int maxProfit=0;
        int currStock=prices[0];
        for(int i=1;i<prices.size();i++){
            if(prices[i]<=currStock){
                maxProfit=0;
                currStock=prices[i];
            }else{
                maxProfit=max(maxProfit,prices[i]-currStock);
            }
        }
        return maxProfit;
    }
};